crates/proof-of-sql-planner/src/expr.rs

```diff
@@ -0,0 +1,424 @@
+
+/// Visit an [`datafusion::expr::Expr`] and return a [`DynProofExpr`]
+pub fn expr_to_proof_expr(expr: &Expr, schema: &DFSchema) -> PlannerResult<DynProofExpr> {
+    match expr {
+        Expr::Column(col) => Ok(DynProofExpr::new_column(column_to_column_ref(col, schema)?)),
+        Expr::BinaryExpr(BinaryExpr { left, right, op }) => {
+            let left_proof_expr = expr_to_proof_expr(left, schema)?;
+            let right_proof_expr = expr_to_proof_expr(right, schema)?;
+            match op {
+                Operator::Eq => Ok(DynProofExpr::try_new_equals(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Lt => Ok(DynProofExpr::try_new_inequality(
+                    left_proof_expr,
+                    right_proof_expr,
+                    true,
+                )?),
+                Operator::Gt => Ok(DynProofExpr::try_new_inequality(
+                    left_proof_expr,
+                    right_proof_expr,
+                    false,
+                )?),
+                Operator::LtEq => Ok(DynProofExpr::try_new_not(
+                    DynProofExpr::try_new_inequality(left_proof_expr, right_proof_expr, false)?,
+                )?),
+                Operator::GtEq => Ok(DynProofExpr::try_new_not(
+                    DynProofExpr::try_new_inequality(left_proof_expr, right_proof_expr, true)?,
+                )?),
+                Operator::Plus => Ok(DynProofExpr::try_new_add(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Minus => Ok(DynProofExpr::try_new_subtract(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Multiply => Ok(DynProofExpr::try_new_multiply(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::And => Ok(DynProofExpr::try_new_and(
+                    left_proof_expr,
+                    right_proof_expr,
+                )?),
+                Operator::Or => Ok(DynProofExpr::try_new_or(left_proof_expr, right_proof_expr)?),
+                _ => Err(PlannerError::UnsupportedBinaryOperator { op: *op }),
+            }
+        }
+        Expr::Literal(val) => Ok(DynProofExpr::new_literal(scalar_value_to_literal_value(
+            val.clone(),
+        )?)),
+        Expr::Not(expr) => {
+            let proof_expr = expr_to_proof_expr(expr, schema)?;
+            Ok(DynProofExpr::try_new_not(proof_expr)?)
+        }
+        _ => Err(PlannerError::UnsupportedLogicalExpression { expr: expr.clone() }),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::df_util::*;
+    use arrow::datatypes::DataType;
+    use datafusion::{common::ScalarValue, logical_expr::expr::Placeholder};
+    use proof_of_sql::{base::database::ColumnType, sql::proof_exprs::utility::*};
+
```

Comment:
NIT:
Adding things like this:
```rust
#[expect(non_snake_case)]
fn COLUMN1_SMALLINT() -> DynProofExpr {
    column_expr(
        Some("namespace"),
        "table_name",
        "column1",
        ColumnType::SmallInt,
    )
}
```
could make these tests way more compact and readable.